kv/dist_sender.go

```diff
@@ -423,16 +423,16 @@ func (ds *DistSender) sendRPC(desc *proto.RangeDescriptor,
 	// Build a slice of replica addresses (if gossiped).
 	var addrs []net.Addr
 	replicaMap := map[string]*proto.Replica{}
-	for i := range replicas {
-		addr, err := ds.gossip.GetNodeIDAddress(replicas[i].NodeID)
+	for _, replica := range replicas {
+		addr, err := ds.gossip.GetNodeIDAddress(replica.NodeID)
 		if err != nil {
 			if log.V(1) {
-				log.Infof("node %d address is not gossiped: %v", replicas[i].NodeID, err)
+				log.Infof("node %d address is not gossiped: %v", replica.NodeID, err)
 			}
 			continue
 		}
 		addrs = append(addrs, addr)
-		replicaMap[addr.String()] = &replicas[i]
+		replicaMap[addr.String()] = &replica
```

Comment:
I think this is incorrect. `&replica` will have the same value throughout the loop. I wonder why no tests are failing.